cpp/src/arrow/compute/kernels/scalar_cast_nested.cc

```diff
@@ -252,6 +353,7 @@ std::vector<std::shared_ptr<CastFunction>> GetNestedCasts() {
   // So is struct
   auto cast_struct = std::make_shared<CastFunction>("cast_struct", Type::STRUCT);
   AddCommonCasts(Type::STRUCT, kOutputTargetType, cast_struct.get());
+  AddStructToStructSubsetCast(cast_struct.get());
```

Comment:
(Just a hacky way to temporarily call the current casting mechanism to test its output). Needs overhaul.

Reply:
(CI fails because of this.)